deployments/ansible/roles/collector/handlers/main.yml

```
@@ -7,28 +7,38 @@
   listen:
     - restart splunk-otel-collector
     - restart td-agent
+  when:
+    - skip_restart | default("false") == "false"
 
 - name: Restart td-agent
   ansible.builtin.service:
     name: td-agent
     state: restarted
-  when: install_fluentd
+  when:
+    - install_fluentd
+    - skip_restart | default("false") == "false"
```

I think this would then expect `skip_restart` to be the string value `"false"` where bools are probably wanted?

```suggestion
    - (skip_restart | default(false) | bool) == false
```